reducer-todo-new/src/components/TodoForm.js

```diff
@@ -0,0 +1,38 @@
+import React, { useState } from "react";
+import { Button, Form } from "semantic-ui-react";
+
+const ToDoForm = (props) => {
+
+    const { dispatch } = props
+    
+    const [ toDoItem, setToDoItem ] = useState({
+        task: '',
+        id: null,
+        completed: false
+    });
+
+    const handleChange = e => {
+     setToDoItem({ ...toDoItem, [e.target.name] : e.target.value})
+    }
```

Comment:
I love that you're using the name attribute.  However, you're only setting 1 field in here.  It might be leaner if you only had a string for your state value, and constructed the object to pass in inside the onSubmit function.